org.jacoco.agent.rt/src/org/jacoco/agent/rt/internal/PreMain.java

```diff
@@ -52,7 +56,49 @@ public static void premain(final String options, final Instrumentation inst)
 
 	private static IRuntime createRuntime(final Instrumentation inst)
 			throws Exception {
+
+		if (redefineJavaBaseModule(inst)) {
+			return new InjectedClassRuntime(Object.class, "$JaCoCo");
+		}
+
 		return ModifiedSystemClassRuntime.createFor(inst, "java/lang/UnknownError");
 	}
 
+	private static boolean redefineJavaBaseModule(
```

Comment:
Maybe add JavaDoc here about Java 9 dependency:

    /**
     * Opens the java.base module for the InjectedClassRuntime when executed on Java 9 JREs or higher.
     *
     * @return <code>true</true> when we're running on Java 9 or higher, otherwise <code>false</code>
     */

Reply:
@marchof done